pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -436,62 +381,115 @@ func (p *PoolManager) MarkVMAsReady(vm *kubevirt.VirtualMachine, latestPersisted
 
 // This function check if there are virtual machines that hits the create
 // mutating webhook but we didn't get the creation event in the controller loop
-// this mean the create was failed by some other mutating or validating webhook
-// so we release the virtual machine
 func (p *PoolManager) vmWaitingCleanupLook() {
 	logger := log.WithName("vmWaitingCleanupLook")
 	c := time.Tick(3 * time.Second)
 	logger.Info("starting cleanup loop for waiting mac addresses")
 	for _ = range c {
-		p.poolMutex.Lock()
-
-		configMap, err := p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Get(context.TODO(), names.WAITING_VMS_CONFIGMAP, metav1.GetOptions{})
-		if err != nil {
-			logger.Error(err, "failed to get config map", "configMapName", names.WAITING_VMS_CONFIGMAP)
-			p.poolMutex.Unlock()
-			continue
-		}
-
-		configMapUpdateNeeded := false
-		if configMap.Data == nil {
-			logger.V(1).Info("the configMap is empty", "configMapName", names.WAITING_VMS_CONFIGMAP, "macPoolMap", p.macPoolMap)
-			p.poolMutex.Unlock()
-			continue
-		}
+		p.healStaleMacEntries(logger)
+	}
+}
 
-		for macAddress, allocationTime := range configMap.Data {
-			t, err := time.Parse(time.RFC3339, allocationTime)
+// healStaleMacEntries looks for stale mac entries, and once find one, heals it by comparing to the real state in the
+// cluster: if the vm still there, and the mac attached to it) then it allocates it, otherwise removes from the macMap
+func (p *PoolManager) healStaleMacEntries(parentLogger logr.Logger) error {
+	p.poolMutex.Lock()
+	defer p.poolMutex.Unlock()
+	logger := parentLogger.WithName("healStaleMacEntries")
+	var macsToRemove []string
+	macsToAlign := map[string]*kubevirt.VirtualMachine{}
+	for macAddress, macEntry := range p.macPoolMap {
+		isEntryStale, err := macEntry.hasExpiredTransaction(p.waitTime)
+		if err == nil && isEntryStale {
+			logger.Info("entry is stale", "macAddress", macAddress, "vmFullName", macEntry.instanceName, "interfaceName", macEntry.macInstanceKey, "stale TS", macEntry.transactionTimestamp)
+
+			var vm *kubevirt.VirtualMachine
+			var err error
+			if macEntry.isDummyEntry() {
+				vm, err = p.recoverVmFromCluster(macAddress)
+			} else {
+				vm, err = p.getvmInstance(macEntry.instanceName)
+			}
 			if err != nil {
-				// TODO: remove the mac from the wait map??
-				logger.Error(err, "failed to parse allocation time")
-				continue
+				if apierrors.IsNotFound(err) {
+					logger.Info("vm no longer exists. Removing mac from pool", "macAddress", macAddress, "entry", macEntry)
+					macsToRemove = append(macsToRemove, macAddress)
+					continue
+				} else {
+					return err
+				}
 			}
+			macsToAlign[macAddress] = vm
+		}
+	}
 
-			logger.Info("data:", "configMapName", names.WAITING_VMS_CONFIGMAP, "configMap.Data", configMap.Data, "macPoolMap", p.macPoolMap)
+	if len(macsToRemove) == 0 && len(macsToAlign) == 0 {
+		return nil
+	}
```

Comment:
Not needed for loop will not even start.

Reply:
yes but it'll prevent the "macMap is self healing" & "macMap is updated" logs.. 
I try that the logs will give a clear picture of what happened..